sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1659,9 +1677,35 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     std::copy_n(PiPlatformsCache->begin(), NumEntries, Platforms);
   }
 
-  if (NumPlatforms)
+  if (NumPlatforms) {
     *NumPlatforms = PiPlatformsCache->size();
 
+    // Check if import user ptr into USM feature has been requested.
+    if (USMHostPtrImport) {
+      // Check if the feature is available.
+      // Since L0 is not platform-specific, just check the first Platform.
+      auto Platform = PiPlatformsCache->front();
+      ze_driver_handle_t driverHandle = Platform->ZeDriver;
+      bool ImportSupported =
+          ZE_CALL_NOCHECK(zeDriverGetExtensionFunctionAddress,
+                          (driverHandle, "zexDriverImportExternalPointer",
+                           reinterpret_cast<void **>(
+                               &zexDriverImportExternalPointer))) == 0;
+      if (ImportSupported) {
+        ZE_CALL_NOCHECK(
+            zeDriverGetExtensionFunctionAddress,
+            (driverHandle, "zexDriverReleaseImportedPointer",
+             reinterpret_cast<void **>(&zexDriverReleaseImportedPointer)));
+        // Turn on hostptr import only if supported and explicitly requested.
```

Comment:
let's make it into an utility function that checks extensions

Reply:
Moved into a separate class.